CMakeLists.txt

```diff
@@ -172,20 +172,28 @@ else()
 endif()
 set_target_properties(foedag-bin PROPERTIES OUTPUT_NAME foedag)
 
-set(TCL_STATIC_LIB libtcl9.0.a)
-set(TCL_STUBB_LIB libtclstub9.0.a)
+set(TCL_STATIC_LIB libtcl8.6.so)
+set(TCL_STUBB_LIB libtclstub8.6.a)
 set(ZLIB_STATIC_LIB libz.a)
 if (MSVC)
-  set(TCL_STUBB_LIB tclstub90.lib)
-  set(TCL_STATIC_LIB tcl90s.lib)
+  set(TCL_STUBB_LIB tclstub86.lib)
+  set(TCL_STATIC_LIB tcl86s.dll)
   set(ZLIB_STATIC_LIB zlib.lib)
+elseif(APPLE) 
+  set(TCL_STATIC_LIB libtcl8.6.dylib)
 endif()
 
-add_library(tcl_static STATIC IMPORTED)
+link_directories(${CMAKE_CURRENT_BINARY_DIR}/lib/)
+
+get_filename_component(buildDirRelFilePath ${TCL_STATIC_LIB}
+                       REALPATH BASE_DIR ${CMAKE_CURRENT_BINARY_DIR}/lib)
+
+add_library(tcl_static SHARED IMPORTED GLOBAL)
+find_library(tcl_static PATHS ${CMAKE_CURRENT_BINARY_DIR}/lib/)
```

Comment:
this will not work. As I understand find_library is used to find libraries that already installed somewhere in the system, like Qt.

Reply:
Strangely enough, it works for Ubuntu, Centos and Macos, but does not work for Windows, although I suspect it's the library name that is wrong.